tests/functional/glusterd/test_custom_xlator.py

```diff
@@ -0,0 +1,327 @@
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import os
+from time import sleep
+
+from glusto.core import Glusto as g
+
+from glustolibs.gluster.exceptions import ExecutionError
+from glustolibs.gluster.gluster_base_class import GlusterBaseClass, runs_on
+from glustolibs.gluster.gluster_init import get_gluster_version
+from glustolibs.gluster.heal_libs import monitor_heal_completion
+from glustolibs.gluster.lib_utils import is_rhel7
+from glustolibs.gluster.rebalance_ops import (rebalance_start,
+                                              wait_for_rebalance_to_complete)
+from glustolibs.gluster.volume_libs import (
+    expand_volume, get_subvols, log_volume_info_and_status,
+    replace_brick_from_volume, wait_for_volume_process_to_be_online)
+from glustolibs.gluster.volume_ops import (get_volume_info, set_volume_options,
+                                           volume_reset, volume_start,
+                                           volume_stop)
+from glustolibs.io.utils import list_all_files_and_dirs_mounts
+
+
+@runs_on([[
+    'distributed', 'arbiter', 'replicated', 'dispersed', 'distributed-arbiter',
+    'distributed-replicated', 'distributed-dispersed'
+], ['glusterfs', 'nfs']])
+# pylint: disable=too-many-statements
+class TestCustomXlator(GlusterBaseClass):
+    '''
+    Description:
+    - Use read-only xlator as a custom xlator to test custom-xlator
+    framework introduced in https://github.com/gluster/glusterfs/pull/1974 and
+    https://github.com/gluster/glusterfs/pull/2371
+
+    Out of scope (or probable bugs):
+    - Options validation of xlator loaded using custom xlator framework
+    - Using existing xlator name for custom xlator in nfs-ganesha
+    - Using custom xlator as parent for another custom/existing xlator
+    '''
+    @classmethod
+    def setUpClass(cls):
+        cls.get_super_method(cls, 'setUpClass')()
+        version = get_gluster_version(cls.mnode).strip()
+        xlator_dir = '/usr/lib64/glusterfs/{0}/xlator'.format(version)
+        cls.usr_dir = os.path.join(xlator_dir, 'user', '')
+
+        # Create 'user' dir and copy read-only xlator as 'ro' and 'posix' as
+        # a custom xlator, 'posix' is used to validate negative scenario
+        cmd = ('mkdir -p {0} && cp {1}/features/read-only.so '
+               '{1}/user/ro.so && cp {1}/user/ro.so {1}/user/posix.so'.format(
+                   cls.usr_dir, xlator_dir))
+
+        out = g.run_parallel(cls.servers, cmd)
+        for host in out:
+            ret, _, _ = out[host]
+            if ret:
+                raise ExecutionError('Unable to create user xlator directory '
+                                     'or failed to copy custom xlator')
+
+    def setUp(self):
+        self.get_super_method(self, 'setUp')()
+
+        # A single mount is enough for all the tests
+        self.mounts = self.mounts[0:1]
```

Comment:
Can't we change it to `self.mounts[0]` as the current code will only have the first element?
```
>>> a = range(1,10)
>>> a
[1, 2, 3, 4, 5, 6, 7, 8, 9]
>>> a[0:1]
[1]
>>> a[0]
1
```

Reply:
It's intentional and the difference can be seen from the example itself.

I actually want a one element *list* which let's me use as it is to `mounts` arg in `setup_volume_and_mount_volume` method

Reply:
I would prefer something like `[self.mounts[0]]`, but it's more of a personal preference not really perticular about it.